src/Processors/Formats/Impl/ArrowBlockInputFormat.cpp

```diff
@@ -31,11 +31,21 @@ Chunk ArrowBlockInputFormat::generate()
     Chunk res;
     const Block & header = getPort().getHeader();
 
-    if (record_batch_current >= record_batch_total)
+    if (!stream && record_batch_current >= record_batch_total)
```

Comment:
Please move this line under the lower if, so that there is a single `if (stream)` block in the function. It will be easier to read.